todo-mvc/src/main/scala/todomvc/TodoMvc.scala

```diff
@@ -178,30 +214,57 @@ object TodoStore:
   def apply(window: Window[IO]): Resource[IO, TodoStore] =
     val key = "todos-calico"
 
+    implicit val encodeFoo: Encoder[(Long, Todo)] = new Encoder[(Long, Todo)] {
+      override def apply(a: (Long, Todo)): Json = {
+        val (id, todo) = a
+        Json.obj(
+          ("id", Json.fromLong(id)),
+          ("title", Json.fromString(todo.text)),
+          ("completed", Json.fromBoolean(todo.completed))
+        )
+      }
+    }
+
+    implicit val decodeFoo: Decoder[(Long, Todo)] = new Decoder[(Long, Todo)] {
+      override def apply(c: HCursor): Result[(Long, Todo)] = for {
+        id <- c.downField("id").as[Long]
+        title <- c.downField("title").as[String]
+        completed <- c.downField("completed").as[Boolean]
+      } yield {
+        (id, Todo(title, completed))
+      }
+    }
+
     for
       mapRef <- SignallingSortedMapRef[IO, Long, Todo].toResource
 
       _ <- Resource.eval {
         OptionT(window.localStorage.getItem(key))
-          .subflatMap(jawn.decode[SortedMap[Long, Todo]](_).toOption)
+          .subflatMap(circe.jawn.decode[List[(Long, Todo)]](_).toOption.map(SortedMap.from))
           .foreachF(mapRef.set(_))
       }
 
-      _ <- window
-        .localStorage
-        .events(window)
-        .foreach {
-          case Storage.Event.Updated(`key`, _, value, _) =>
-            jawn.decode[SortedMap[Long, Todo]](value).foldMapM(mapRef.set(_))
-          case _ => IO.unit
-        }
-        .compile
-        .drain
-        .background
+//      _ <- window
+//        .localStorage
+//        .events(window)
+//        .foreach {
+//          case Storage.Event.Updated(`key`, _, value, _) =>
+//            jawn.decode[SortedMap[Long, Todo]](value).foldMapM(mapRef.set(_))
+//          case _ => IO.unit
+//        }
+//        .compile
+//        .drain
+//        .background
```

Comment:
Why is it commented?

Reply:
I did not see the use case for that, it works fine without.
I uncommented it anyway.

Reply:
> I did not see the use case for that, it works fine without.

Try opening two tabs to the TodoMVC page at the same time :)